### DISCUSSION TITLE
Publishing to pub.dev

### DISCUSSION THREAD

**Discussion by piekarskipiotr** [codemagic-ci-cd/codemagic-docs#1802] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

  Use the docs example script 
  ```
  environment:
    vars:
      PUB_DEV_CREDENTIALS

  scripts:
    - name: Publish to pub.dev
      script: | 
        echo $PUB_DEV_CREDENTIALS > "$FLUTTER_ROOT/.pub-cache/credentials.json"
        flutter pub publish --dry-run
        flutter pub publish -f
```
### Expected results

Should publish to pub.dev without any interaction.

### Actual results

Getting:
/Users/builder/programs/flutter/.pub-cache/credentials.json: No such file or directory

Pub needs your authorization to upload packages on your behalf.
In a web browser, go to <link>
Then click "Allow access".

Waiting for your authorization...

**Comment by rafeysup93** [comment, marked as answer]
Hi @piekarskipiotr,

The error message `No such file or directory` would mean the file's path isn't correct. I suggest you review the file path by remotely accessing the machine to see if the `credentials.json` file exists in the mentioned location. 

Please go through this [document](https://docs.codemagic.io/troubleshooting/accessing-builder-machine-via-ssh/) for steps to access the build machine remotely.

**Comment by piekarskipiotr** [comment]
Thanks💙 Connecting to build machine via ssh was really helpful, it's working now!
